### DISCUSSION TITLE
Making a custom type be serializable by itself in v2

### DISCUSSION THREAD

**Discussion by ashleney** [pydantic/pydantic#6351] [Question] [answered]
In v1 I could simply define a `__get_validators__`, now the documentation is suggesting I specify this outside the class and I can't find any documentation for how to do this inside.

My v1 code:
```py
@classmethod
def __get_validators__(cls) -> typing.Iterator[typing.Callable[..., typing.Any]]:
    yield lambda i: cls(i)
```

Any class this is attached to should be casted when serialized.

**Comment by adriangb** [comment, marked as answer]
See https://docs.pydantic.dev/latest/usage/types/custom/#creating-custom-classes-using-__get_pydantic_core_schema__
